provisioning_client/adapters/hsm_client_riot.c

```diff
@@ -551,6 +548,15 @@ void hsm_client_riot_destroy(HSM_CLIENT_HANDLE handle)
         /* Codes_SRS_HSM_CLIENT_RIOT_07_008: [ hsm_client_riot_destroy shall free the HSM_CLIENT_HANDLE instance. ] */
         free(x509_client->certificate_common_name);
         /* Codes_SRS_HSM_CLIENT_RIOT_07_009: [ hsm_client_riot_destroy shall free all resources allocated in this module. ] */
+        mbedtls_ecp_point_free(&x509_client->ca_root_pub);
+        mbedtls_mpi_free(&x509_client->ca_root_priv);
+
+        mbedtls_ecp_point_free(&x509_client->device_id_pub);
```

Comment:
Is this the right function?
[If you search for `mbedtls_ecp_point_free` in these docs](https://tls.mbed.org/api/ecp_8h.html#a87004f74814cc59a9ef757aea444098d) the type to pass here is `mbedtls_ecp_point*` and device_id_pub I think is `RIOT_ECC_PUBLIC`

Reply:
Any pub struct is an ecp_point and any private is an mbedtls_mpi struct